scene/gui/code_edit.cpp

```diff
@@ -101,6 +103,71 @@ void CodeEdit::_notification(int p_what) {
 				}
 			}
 
+			/* Codeblock Guidelines */
+			if (codeblock_guidelines_enabled) {
+				const int xmargin_beg = style_normal->get_margin(SIDE_LEFT) + get_total_gutter_width();
+				const float space_width = font->get_char_size(' ', font_size).width;
+				const double v_scroll = get_v_scroll();
+				const double h_scroll = get_h_scroll();
+				Vector<int> block_ends;
+				for (int i = 0; i < get_line_count(); i++) {
+					if (_is_line_hidden(i)) {
+						continue;
+					}
+
+					if (can_fold_line(i)) {
+						const int block_start = i; // This is the Fold Gutter line.
+						const int block_end = _get_fold_line_ending(block_start);
+
+						if (block_end <= 0) {
+							continue;
+						}
+
+						const int indent_level_start = get_indent_level(i);
+						const int indent_level_inner = get_indent_level(i + 1);
+
+						// Check if this codeblock contains the caret.
+						Color color = codeblock_guideline_color;
+						const int caret_idx = get_caret_line();
+
+						if (caret_idx > block_start &&
+								caret_idx <= block_end &&
+								get_indent_level(caret_idx) == indent_level_inner) {
+							color = codeblock_guideline_active_color;
+						}
+
+						float indent_guide_x = (indent_level_start * space_width + space_width / 2) + xmargin_beg - h_scroll;
+
+						int skipped_lines_to_start = block_start - get_visible_line_count_in_range(0, block_start);
+						int skipped_lines_to_end = block_end - get_visible_line_count_in_range(0, block_end);
+						const double visible_indent_start = (block_start - v_scroll - skipped_lines_to_start);
+						const double visible_indent_end = (block_end - v_scroll - skipped_lines_to_end);
+
+						// Stack multiple guidelines.
+						float offset_y = MIN(block_ends.count(block_end) * 2.0, font->get_height(font_size) / 2.0);
+
+						// Vertical line to the end.
+						Point2 point_start = Point2(indent_guide_x, row_height * visible_indent_start);
+						Point2 point_end = Point2(indent_guide_x, row_height * visible_indent_end - offset_y);
+
+						RenderingServer::get_singleton()->canvas_item_add_line(ci, point_start, point_end, color, 1.0);
+
+						// Horizontal guideline starting from the end,
+						// Drawn whenever a parentesis underneath is unavailable, or already taken by a higher guideline.
+						String line_below = get_line(block_end + 1).strip_edges();
+						bool bottom_begins_with_parenthesis = !(line_below.begins_with(")") || line_below.begins_with("]") || line_below.begins_with("}"));
```

Comment:
Can we use `brace_completion` code keys for this?

Reply:
It is possible, yeah.